lib/services/application-service.ts

```diff
@@ -0,0 +1,97 @@
+import util from "../shared/util";
+
+export class ApplicationService implements IApplicationService {
+    constructor(private $pageService: IPageService) { }
+
+    getPages(): Promise<Array<any>> {
+        return this.$pageService.getPages();
+    }
+
+    getFlavor(appPath: string): Promise<string> {
+        return new Promise((resolve, reject) => {
+            this.getAppPath(appPath)
+                .then((validPath: string) => {
+                    return this.readAppPackageJson(validPath);
+                })
+                .then((packageJson: string) => {
+                    const ng = packageJson.indexOf("nativescript-angular") > -1;
+                    const ts = packageJson.indexOf("nativescript-dev-typescript") > -1;
+
+                    switch (true) {
+                        case ng:
+                            resolve("Angular & TypeScript");
+                            break;
+                        case ts:
+                            resolve("TypeScript");
+                            break;
+                        default:
+                            resolve("JavaScript");
+
+                    }
+                })
+                .catch((promiseError: any) => {
+                    reject(promiseError);
+                });
+        });
+    }
+
+    addPage(pageName: string, appPath: string, pageTemplate: any): Promise<string> {
+        return this.$pageService.addPage(pageName, appPath, pageTemplate);
+    }
+
+    private getAppPath(appPath: string) {
```

Comment:
This method name is bugging me -- we pass appPath as a parameter and get appPath as output? There must be a more descriptive name for the method (or the parameter)?

Maybe getNormalizedAppPath(...) ?